azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/SetupViewModel.kt

```diff
@@ -34,22 +35,32 @@ internal class SetupViewModel(
     }
 
     fun exitComposite() {
+        // double check here if we need both the action to execute
         dispatchAction(action = CallingAction.CallEndRequested())
         dispatchAction(action = NavigationAction.Exit())
     }
 
     override fun init(coroutineScope: CoroutineScope) {
         val state = store.getCurrentState()
+
+        if (store.getCurrentState().localParticipantState.initialCallJoinState.startWithMicrophoneOn) {
```

Comment:
In one call it doesn't get called more than once. If it were, then the whole purpose of setup viewmodel init is wrong. Activity instance is recycled once the composite exits or callEnd is called. I think after that, while joining a call, we want these codes to be called with the config being set by users. 
Otherwise, whole setupViewModel init doesn't make sense. Everything would be initialized multiple times.